### DISCUSSION TITLE
Reading packet length from InputStream or forwarding bytes directly to WebSocket

### DISCUSSION THREAD

**Discussion by mfurmanit** [kokorin/Jaffree#253] [Q&A]
Awesome library, I'm using Jaffree to convert and display my RTSP stream using FFmpeg and JSMpeg.
I have a problem with memory optimization and consumption though.
I described the problem in details at [this thread](https://stackoverflow.com/questions/70200009/forward-stdout-bytes-from-process-to-websocket-or-read-current-inputstream-lengt) in stackoverflow.

If I have to describe a problem in this thread directly, instead of referring to stackoverlow, just let me know.

Can I somehow predict length of a packet that was sent to stdOut from FFmpeg?
Or maybe somehow redirect stdOut directly to WebSocket?

I will be very grateful for any help and information if it is possible.

**Comment by kokorin** [comment]
Have you checked `PipeOutput` and `ChannelOutput`? They do provide byte-level access to output stream.

**Reply by mfurmanit** [reply]
@kokorin I've updated my question on stackoverflow. Let me know if you prefer to communicate here or there.

I've implemented it using `PipeOutput` according to your answer, but it has the exact same limitations. Under the hood `PipeOutput` just copies stream content to `new byte[]` initialized with predefined buffer size. So i get outputs like this:

```
Array length: 262144, actual length: 9588
Array length: 262144, actual length: 4700
Array length: 262144, actual length: 9024
Array length: 262144, actual length: 10152
Array length: 262144, actual length: 18048
```

I need to send arrays with actual length (using WebSocket) rather than array length. The reason is the way that JSMpeg library works.
